docs/source/comparison.rst

```diff
@@ -0,0 +1,69 @@
++-------+-----------------------------+-------------------+------------------------+-------------------+----------------------------------------------------+
+| NNs   | Reverse-mode AD             | Y                 | Y                      | Y                 | Y                                                  |
+|       +-----------------------------+-------------------+------------------------+-------------------+----------------------------------------------------+
+|       | Basic RNN support           | Y                 | Y                      | Y (nnx)           | `#2033 <https://github.com/BVLC/caffe/pull/2033>`_ |
+|       +-----------------------------+-------------------+------------------------+-------------------+----------------------------------------------------+
+|       | Variable-length loops       | Y                 | Y (scan)               |                   |                                                    |
+|       +-----------------------------+-------------------+------------------------+-------------------+----------------------------------------------------+
+|       | Stateful RNNs               | Y                 |                        |                   |                                                    |
+|       +-----------------------------+-------------------+------------------------+-------------------+----------------------------------------------------+
+|       | Per-batch architectures     | Y                 |                        |                   |                                                    |
++-------+-----------------------------+-------------------+------------------------+-------------------+----------------------------------------------------+
+| Perf  | CUDA support                | Y                 | Y                      | Y                 | Y                                                  |
+|       +-----------------------------+-------------------+------------------------+-------------------+----------------------------------------------------+
+|       | cuDNN support               | Y                 | Y                      | Y (cudnn.torch)   | Y                                                  |
+|       +-----------------------------+-------------------+------------------------+-------------------+----------------------------------------------------+
+|       | FFT-based convolution       |                   | Y                      | Y (fbcunn)        | `#544 <https://github.com/BVLC/caffe/pull/544>`_   |
+|       +-----------------------------+-------------------+------------------------+-------------------+----------------------------------------------------+
+|       | CPU/GPU generic coding      | [1]_              | [4]_                   | Y                 |                                                    |
```

Comment:
I believe what we want to emphasize is whether we can write code that can usable both CPU and GPU mode. It would be better to add some explanations for it.